components/jaeger-collector.libsonnet

```diff
@@ -3,110 +3,150 @@ local service = k.core.v1.service;
 local jaegerAgent = import './jaeger-agent.libsonnet';
 
 {
-  jaeger+:: {
-    local j = self,
+  local _self = self,
```

Comment:
in other places we call this what the object is called, which would be `jaeger` in this case.. I find `_self` to be confusing as `self` is relative to the object but `_self` wouldn't be.

Reply:
why `_self` isn't relative to the object?

I find it confusing the other way around. for example, in many places, I see `obs` and it is very difficult to trace down what does it point to.

Reply:
it's always the top level object that's being exported, I think the pattern is discussable, but if we change it we should do it consistently and not do it in one place but not the rest

Reply:
yes I see what you mean, I am still trying different things so let me think about this a bit again.

Reply:
(if I wasn't clear I think we should not change the style for all others in this PR as well, in this PR we should stay consistent with the rest and then in a follow up we can discuss style changes for everything)